diff --git a/src/operations/config/is_federated/runner.rs b/src/operations/config/is_federated/runner.rs
index 35084ce..42849a6 100644
--- a/src/operations/config/is_federated/runner.rs
+++ b/src/operations/config/is_federated/runner.rs
@@ -10,7 +10,7 @@ use graphql_client::*;
 // Both json and the GraphQL schema language are supported as sources for the schema
 #[graphql(
     query_path = "src/operations/config/is_federated/is_federated_query.graphql",
-    schema_path = ".schema/schema.graphql",
+    schema_path = "../../third_party/rover_client/.schema/schema.graphql",
     response_derives = "PartialEq, Debug, Serialize, Deserialize",
     deprecated = "warn"
 )]
diff --git a/src/operations/config/who_am_i/runner.rs b/src/operations/config/who_am_i/runner.rs
index 0974cd4..6ad25ba 100644
--- a/src/operations/config/who_am_i/runner.rs
+++ b/src/operations/config/who_am_i/runner.rs
@@ -14,7 +14,7 @@ use graphql_client::*;
 // Both json and the GraphQL schema language are supported as sources for the schema
 #[graphql(
     query_path = "src/operations/config/who_am_i/who_am_i_query.graphql",
-    schema_path = ".schema/schema.graphql",
+    schema_path = "../../third_party/rover_client/.schema/schema.graphql",
     response_derives = "PartialEq, Debug, Serialize, Deserialize",
     deprecated = "warn"
 )]
diff --git a/src/operations/graph/check/runner.rs b/src/operations/graph/check/runner.rs
index 37a9ddc..c1ba06c 100644
--- a/src/operations/graph/check/runner.rs
+++ b/src/operations/graph/check/runner.rs
@@ -11,7 +11,7 @@ type Timestamp = String;
 // Both json and the GraphQL schema language are supported as sources for the schema
 #[graphql(
     query_path = "src/operations/graph/check/check_mutation.graphql",
-    schema_path = ".schema/schema.graphql",
+    schema_path = "../../third_party/rover_client/.schema/schema.graphql",
     response_derives = "PartialEq, Debug, Serialize, Deserialize",
     deprecated = "warn"
 )]
diff --git a/src/operations/graph/delete/runner.rs b/src/operations/graph/delete/runner.rs
index 726a8a2..a0cfed3 100644
--- a/src/operations/graph/delete/runner.rs
+++ b/src/operations/graph/delete/runner.rs
@@ -10,7 +10,7 @@ use graphql_client::*;
 #[derive(GraphQLQuery)]
 #[graphql(
     query_path = "src/operations/graph/delete/delete_mutation.graphql",
-    schema_path = ".schema/schema.graphql",
+    schema_path = "../../third_party/rover_client/.schema/schema.graphql",
     response_derives = "PartialEq, Debug, Serialize, Deserialize",
     deprecated = "warn"
 )]
diff --git a/src/operations/graph/fetch/runner.rs b/src/operations/graph/fetch/runner.rs
index fc72a3b..230eca8 100644
--- a/src/operations/graph/fetch/runner.rs
+++ b/src/operations/graph/fetch/runner.rs
@@ -14,7 +14,7 @@ type GraphQLDocument = String;
 // Both json and the GraphQL schema language are supported as sources for the schema
 #[graphql(
     query_path = "src/operations/graph/fetch/fetch_query.graphql",
-    schema_path = ".schema/schema.graphql",
+    schema_path = "../../third_party/rover_client/.schema/schema.graphql",
     response_derives = "PartialEq, Debug, Serialize, Deserialize",
     deprecated = "warn"
 )]
diff --git a/src/operations/graph/publish/runner.rs b/src/operations/graph/publish/runner.rs
index 4ff0fba..73f1a0b 100644
--- a/src/operations/graph/publish/runner.rs
+++ b/src/operations/graph/publish/runner.rs
@@ -10,7 +10,7 @@ use graphql_client::*;
 // Both json and the GraphQL schema language are supported as sources for the schema
 #[graphql(
     query_path = "src/operations/graph/publish/publish_mutation.graphql",
-    schema_path = ".schema/schema.graphql",
+    schema_path = "../../third_party/rover_client/.schema/schema.graphql",
     response_derives = "PartialEq, Debug, Serialize, Deserialize",
     deprecated = "warn"
 )]
diff --git a/src/operations/graph/variant/runner.rs b/src/operations/graph/variant/runner.rs
index 5a3d4d4..88862fb 100644
--- a/src/operations/graph/variant/runner.rs
+++ b/src/operations/graph/variant/runner.rs
@@ -7,7 +7,7 @@ use graphql_client::*;
 #[derive(GraphQLQuery)]
 #[graphql(
     query_path = "src/operations/graph/variant/variant_query.graphql",
-    schema_path = ".schema/schema.graphql",
+    schema_path = "../../third_party/rover_client/.schema/schema.graphql",
     response_derives = "PartialEq, Debug, Serialize, Deserialize",
     deprecated = "warn"
 )]
diff --git a/src/operations/subgraph/check/runner.rs b/src/operations/subgraph/check/runner.rs
index d0b47d0..8a73080 100644
--- a/src/operations/subgraph/check/runner.rs
+++ b/src/operations/subgraph/check/runner.rs
@@ -17,7 +17,7 @@ type Timestamp = String;
 // Both json and the GraphQL schema language are supported as sources for the schema
 #[graphql(
     query_path = "src/operations/subgraph/check/check_mutation.graphql",
-    schema_path = ".schema/schema.graphql",
+    schema_path = "../../third_party/rover_client/.schema/schema.graphql",
     response_derives = "PartialEq, Debug, Serialize, Deserialize",
     deprecated = "warn"
 )]
diff --git a/src/operations/subgraph/delete/runner.rs b/src/operations/subgraph/delete/runner.rs
index e80b140..ac9b744 100644
--- a/src/operations/subgraph/delete/runner.rs
+++ b/src/operations/subgraph/delete/runner.rs
@@ -12,7 +12,7 @@ use graphql_client::*;
 // Both json and the GraphQL schema language are supported as sources for the schema
 #[graphql(
     query_path = "src/operations/subgraph/delete/delete_mutation.graphql",
-    schema_path = ".schema/schema.graphql",
+    schema_path = "../../third_party/rover_client/.schema/schema.graphql",
     response_derives = "PartialEq, Debug, Serialize, Deserialize",
     deprecated = "warn"
 )]
diff --git a/src/operations/subgraph/fetch/runner.rs b/src/operations/subgraph/fetch/runner.rs
index 380421f..2d0306f 100644
--- a/src/operations/subgraph/fetch/runner.rs
+++ b/src/operations/subgraph/fetch/runner.rs
@@ -10,7 +10,7 @@ use graphql_client::*;
 // Both json and the GraphQL schema language are supported as sources for the schema
 #[graphql(
     query_path = "src/operations/subgraph/fetch/fetch_query.graphql",
-    schema_path = ".schema/schema.graphql",
+    schema_path = "../../third_party/rover_client/.schema/schema.graphql",
     response_derives = "PartialEq, Debug, Serialize, Deserialize",
     deprecated = "warn"
 )]
diff --git a/src/operations/subgraph/list/runner.rs b/src/operations/subgraph/list/runner.rs
index 2b99127..706f5b3 100644
--- a/src/operations/subgraph/list/runner.rs
+++ b/src/operations/subgraph/list/runner.rs
@@ -11,7 +11,7 @@ type Timestamp = String;
 // Both json and the GraphQL schema language are supported as sources for the schema
 #[graphql(
     query_path = "src/operations/subgraph/list/list_query.graphql",
-    schema_path = ".schema/schema.graphql",
+    schema_path = "../../third_party/rover_client/.schema/schema.graphql",
     response_derives = "PartialEq, Debug, Serialize, Deserialize",
     deprecated = "warn"
 )]
diff --git a/src/operations/subgraph/publish/runner.rs b/src/operations/subgraph/publish/runner.rs
index 9364547..3edab23 100644
--- a/src/operations/subgraph/publish/runner.rs
+++ b/src/operations/subgraph/publish/runner.rs
@@ -17,7 +17,7 @@ use apollo_federation_types::{BuildError, BuildErrors};
 // Both json and the GraphQL schema language are supported as sources for the schema
 #[graphql(
     query_path = "src/operations/subgraph/publish/publish_mutation.graphql",
-    schema_path = ".schema/schema.graphql",
+    schema_path = "../../third_party/rover_client/.schema/schema.graphql",
     response_derives = "PartialEq, Debug, Serialize, Deserialize",
     deprecated = "warn"
 )]
diff --git a/src/operations/supergraph/fetch/runner.rs b/src/operations/supergraph/fetch/runner.rs
index 4c935b6..5018da1 100644
--- a/src/operations/supergraph/fetch/runner.rs
+++ b/src/operations/supergraph/fetch/runner.rs
@@ -16,7 +16,7 @@ type GraphQLDocument = String;
 // Both json and the GraphQL schema language are supported as sources for the schema
 #[graphql(
     query_path = "src/operations/supergraph/fetch/fetch_query.graphql",
-    schema_path = ".schema/schema.graphql",
+    schema_path = "../../third_party/rover_client/.schema/schema.graphql",
     response_derives = "PartialEq, Debug, Serialize, Deserialize",
     deprecated = "warn"
 )]
--
2.25.1

